tools/sdk-generation-pipeline/ci.yml

```diff
@@ -0,0 +1,147 @@
+# Node.js
+# Build a general Node.js project with npm.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+
+# Node.js
+# Build a general Node.js project with npm.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+
+trigger:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/sdk-generation-pipeline
+
+pr:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/sdk-generation-pipeline
+
+variables:
+  - template: ../../eng/pipelines/templates/variables/globals.yml
+  - name: NodeVersion
+    value: '14.x'
+  - name: VAR_ARTIFACT_NAME
+    value: 'drop'
+  - name: VAR_BUILD_ARTIFACT_STAGING_DIRECTORY
+    value: $(Build.ArtifactStagingDirectory)
+
+stages:
+  - stage: InstallAndBuild
+    jobs:
+      - job: Build
+        strategy:
+          matrix:
+            linux:
+              imageName: 'ubuntu-latest'
```

Comment:
Should we use the 1es hosted pools that we maintain for windows and mac, which have higher scaling?

```
          pool:
            name: 'azsdk-pool-mms-ubuntu-2004-general'
            vmImage: 'MMSUbuntu20.04'
```

```
          pool:
            name: 'azsdk-pool-mms-win-2019-general'
            vmImage: 'MMS2019'
```

Reply:
I'm not sure which pool should be used here and I copy them from `mock-service-host`. I think the pool you provide is ubuntu and windows, and there is no mac. Could you also provide the pool of mac? I will update it later. Thanks
ps: I will merge this PR firstly because I don't want to block others' work.